model/riscv_insts_fext.sail

```diff
@@ -518,7 +550,14 @@ function clause execute (F_BIN_RM_TYPE_S(rs2, rs1, rm, rd, op)) = {
     FDIV_S  => riscv_f32Div (rm_3b, rs1_val_32b, rs2_val_32b)
   };
   write_fflags(fflags);
-  F(rd) = nan_box (rd_val_32b);
+  
+
+  if   ( sys_enable_zfinx() & sizeof(xlen) == 64 )
+  then X(rd) = x_nan_box (rd_val_32b)
+  else if ( sys_enable_zfinx() & sizeof(xlen) == 32 )
+  then X(rd) = rd_val_32b
+  else F(rd) = nan_box (rd_val_32b);
```

Comment:
This paragraph of code appears 8 times in this file. Surely there's a better way.